agent/api/eni/eni.go

```diff
@@ -208,8 +211,13 @@ func (eni *ENI) GetLinkName() string {
 			return ""
 		}
 		// Iterate over the list and find the interface with the ENI's MAC address.
+		eniMac, err := net.ParseMAC(eni.MacAddress)
+		if err != nil {
+			seelog.Errorf("Failed to parse ENI MAC: %s with error: %v", eni.MacAddress, err)
+			return ""
+		}
 		for _, iface := range ifaces {
-			if strings.EqualFold(eni.MacAddress, iface.HardwareAddr.String()) {
+			if bytes.EqualFold(eniMac, iface.HardwareAddr) {
```

Comment:
Would it be helpful to check if LinkName is non-empty prior to return?

Reply:
Good call resolved.